### DISCUSSION TITLE
Save access token on cookies

### DISCUSSION THREAD

**Discussion by paulospiguel** [kinngh/shopify-node-mongodb-next-app#39] [Q&A]
Hello, how can we set the cookies for this app within the Shopify platform? I have a use case where I need to save an access token inside the cookies to make future API calls but I don't want to make the request every time to get that access token. Do you know how I can implement this?

**Comment by kinngh** [comment]
1. Very bad idea
2. Please don't do it
3. I do know but if you have to open a discussion and ask someone, you should go back to doing baseline tutorials on how Next and Cookies work and do it yourself
4. I'd love to know more about your use case where you have to store highly sensitive information openly in a cookie to make `future` API calls
5. It's an extremely terrible idea to do this

**Reply by paulospiguel** [reply]
My question is. I need to make a call to an API to search for products. However, first I need to do a search to get an access token to send in the autorization call in the headers. But my question is what strategy would I use to make these product calls without having to search for a new access token each time, but only when it has expired? Maybe I could store it either in a cookie or in a database and check if it's expired. If so, then it fetches a new one (refresh_token), otherwise it uses the same one. Do you see how to do this differently?

**Reply by kinngh** [reply]
Why not use Resource Picker from App bridge?

**Reply by paulospiguel** [reply]
I don't know Resource Picker. I need to research. Thanks for the tip!

**Reply by kinngh** [reply]
https://shopify.dev/docs/api/app-bridge-library/reference/resource-picker
